### PR TITLE
destroy operator and code cleanup

### PR DESCRIPTION
**Pull request by msenmurugan**
* destroy operator if there is no other custom resource owned by it
* removed cmd/operator unwanted code
* cleaned up the Makefile and .gitignore
* added code to remove the owner label in the namespace
* during delete operation, remove exposed services and pvcs if exist

### CODE REVIEW

**Comment by operator-r2d2** [comment]
[APPROVALNOTIFIER] This PR is **APPROVED**

This pull-request has been approved by: *<a href="https://github.com/blackducksoftware/synopsys-operator/pull/1326#" title="Author self-approved">msenmurugan</a>*

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=blackducksoftware%2Fsynopsys-operator).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details >
Needs approval from an approver in each of these files:

- ~~[OWNERS](https://github.com/blackducksoftware/synopsys-operator/blob/develop/OWNERS)~~ [msenmurugan]

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

File: pkg/synopsysctl/cmd_destroy.go [hunk 1]
```diff
@@ -323,8 +216,100 @@ func deleteCrd(crd string, namespace string) {
...
+			err := util.DeleteClusterRole(kubeClient, clusterRole)
+			if err != nil {
+				log.Errorf("unable to delete cluster role '%s' due to %+v", clusterRole, err)
+			}
+		}
+	}
+}
+
+// isDeleteOperatorResources check whether there is any custom resources are existing
+func isDeleteOperatorResources(crdNamespace string, crdNames []string) (bool, error) {
```
Comment (mphammer) [inline pkg/synopsysctl/cmd_destroy.go:285]: Can we rename this function to something like "CustomResourceInstancesExist"? I got a little confused

**Comment by mphammer** [comment]
Can we rename https://github.com/blackducksoftware/synopsys-operator/pull/1326/files#diff-5eb677ff0a081c6b1e6c94421013f70cR129 "isDeleteCRD" to something like "CRDIsNotUsedInAnyNamespaces"

**Comment by operator-r2d2** [comment]
[APPROVALNOTIFIER] This PR is **APPROVED**

This pull-request has been approved by: *<a href="https://github.com/blackducksoftware/synopsys-operator/pull/1326#pullrequestreview-392184058" title="Approved">mphammer</a>*, *<a href="https://github.com/blackducksoftware/synopsys-operator/pull/1326#" title="Author self-approved">msenmurugan</a>*

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=blackducksoftware%2Fsynopsys-operator).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details >
Needs approval from an approver in each of these files:

- ~~[OWNERS](https://github.com/blackducksoftware/synopsys-operator/blob/develop/OWNERS)~~ [mphammer,msenmurugan]

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Comment by operator-r2d2** [comment]
[APPROVALNOTIFIER] This PR is **APPROVED**

This pull-request has been approved by: *<a href="https://github.com/blackducksoftware/synopsys-operator/pull/1326#pullrequestreview-392280194" title="Approved">jdartigalongue</a>*, *<a href="https://github.com/blackducksoftware/synopsys-operator/pull/1326#pullrequestreview-392184058" title="Approved">mphammer</a>*, *<a href="https://github.com/blackducksoftware/synopsys-operator/pull/1326#" title="Author self-approved">msenmurugan</a>*

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=blackducksoftware%2Fsynopsys-operator).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details >
Needs approval from an approver in each of these files:

- ~~[OWNERS](https://github.com/blackducksoftware/synopsys-operator/blob/develop/OWNERS)~~ [jdartigalongue,mphammer,msenmurugan]

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Pull request merged**